pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -516,6 +519,7 @@ func (l *LibvirtDomainManager) SetGuestTime(vmi *v1.VirtualMachineInstance) erro
 		log.Log.Object(vmi).Reason(err).Error("failed to sync guest time")
 		return err
 	}
+	defer dom.Free()
 	// try setting the guest time. Fail early if agent is not configured.
 	l.setGuestTime(vmi, dom)
```

Comment:
setGuestTime is executing a go routine that has this domain passed into it.

Reply:
Thanks. Done.